code/test/app.js

```diff
@@ -81,54 +70,393 @@ const executeAppTransaction = async ({
+  // SignatureManager.sol revokeHandler
+  // function revokeHandler(SignatureTracker storage st, address contractAddress, uint256 tokenId, address payable dest, uint256 amount) public {
+  //     function revoke(SignatureTracker storage st, bytes32 hash, bytes32 signature) public returns (bool){
+  // Revoke Logic
+  // contractAddress (if != address(0) then revoke everything base on time if passed )
+  // dest = expireAt (if > 0 then revoke all signatures before this time)
+  // op.tokenId = hash
+  // op.amount = signature
+  it('AP-BASIC-20 REVOKE: must be able to revoke a signature', async () => {
+    // Begin Tests
+    let testTime = Date.now()
+
+    testTime = await TestUtil.bumpTestTime(testTime, 60)
+
+    const hexData = ONEUtil.abi.encodeParameters(['address', 'uint16', 'bytes'], [alice.wallet.address, ONEConstants.OperationType.SIGN, new Uint8Array()])
+    const messageHash = ONEUtil.keccak('hello world')
+    const signature = ONEUtil.keccak('awesome signature')
+    // const expiryAtDate = Math.floor(((testTime + 30000) / 1000)) // 5 mins after testTime
+    // let expiryAtBytes = new BN(expiryAtDate).toArrayLike(Uint8Array, 'be', 4)
+    let expiryAtBytes = new BN(0xffffffff).toArrayLike(Uint8Array, 'be', 4)
+    let encodedExpiryAt = new Uint8Array(20)
+    encodedExpiryAt.set(expiryAtBytes)
+    let data = ONEUtil.hexStringToBytes(hexData)
+    // create a signature so that we can revoke it
+    let { tx, currentState } = await executeAppTransaction(
+      {
+        ...NullOperationParams, // Default all fields to Null values than override
+        walletInfo: alice,
+        operationType: ONEConstants.OperationType.SIGN,
+        tokenId: new BN(messageHash).toString(),
+        dest: ONEUtil.hexString(encodedExpiryAt),
+        amount: new BN(signature).toString(),
+        data,
+        testTime
+      }
+    )
+    // Validate succesful event emitted
+    TestUtil.validateEvent({ tx, expectedEvent: 'SignatureAuthorized' })
+
+    // Alice items that have changed -signatures
+    state = await TestUtil.validateOpsStateMutation({ wallet: alice.wallet, state })
+    // check alice signatures have changed by getting the current values and overriding with the expected hash and signature
+    let expectedSignatures = await TestUtil.getSignaturesParsed(alice.wallet)
+    expectedSignatures[0].hash = ONEUtil.hexString(messageHash)
+    expectedSignatures[0].signature = ONEUtil.hexString(signature)
+    state.signatures = await TestUtil.validateSignaturesMutation({ expectedSignatures, wallet: alice.wallet })
+    await TestUtil.assertStateEqual(state, currentState)
+
+    // now revoke the signature
+    testTime = await TestUtil.bumpTestTime(testTime, 60)
+    let { currentState: currentStateRevoked } = await executeAppTransaction(
+      {
+        ...NullOperationParams, // Default all fields to Null values than override
+        walletInfo: alice,
+        operationType: ONEConstants.OperationType.REVOKE,
```

Comment:
revoke is not working here because `dest` (i.e. beforeTime) is 0, and `tokenId` and `amount (which represents the message hash and the signature) are also empty. So nothing is revoked

Reply:
Thanks 🙏 
Have just updated all the SignatureExpired tests and also tested revoke by DATE and SIGNATURE